Docs Concurrency update

tomasonjo opened the pull request:
I have found few things, that might be changed, so that as default we use as many threads as possible...

algo.scc.multistep uses:

```
configuration.getNumber("concurrency", 1).intValue()
```

algo.scc.forwardBackward uses:

```
 configuration.getConcurrency(1))
```

And correct me if I am wrong `algo.betweenness.exp1` seems to have hard-coded 4 threads:

```
 ParallelUtil.iterateParallel(executorService, nodeCount, 4, i -> {
            d.set(i, -1);
            sigma.set(i, 0);
            offsets.set(i, 0);
        });
```
when i compare it to:

```
ParallelUtil.iterateParallel(executorService, nodeCount, Pools.DEFAULT_CONCURRENCY, i -> {
                centrality.set(i, centrality.get(i) / 2.0);
            });
```

I can easily change `algo.scc.*` to default to that it will use maximum threads possible
```
configuration.getConcurrency()
```

jexp commented:
@knutwalker @mknblch ping.

jexp commented:
@mknblch was there a reason why we default to 1 for these procedures? Might be quite confusing for the users? Wouldn't it be more consistent to use the same concurrency approach everywhere?

jexp commented:
Also let's change the documented concurrency to 4 not 8. Most people  don't have that many cores on their machines.

tomasonjo commented:
I talked to @mknblch and he said there is no problem if I change the default for `algo.scc.*` to default `configuration.getConcurrency()`. There remains only betweenness then which runs on 1 thread as default.

tomasonjo commented:
updated

mknblch commented:
We should use `Pools.DEFAULT_CONCURRENCY` or just `conf.getConcurrency()` without argument in the procedures itself. In Descriptions i would suggest 4. 

If there are hardcoded values somewhere it's probably a leftover of some tests.

tomasonjo commented:
I also found `configuration.getInt("concurrency", 4)` in `algo.shortestPath.deltaStepping.stream`, that I changed to `conf.getConcurrency()`.

To recap what I changed:

Changed concurrency settings to use `conf.getConcurrency()`.for :
`
- algo.shortestPath.deltaStepping.stream
- algo.scc.multistep
- algo.scc.multistep.stream
- algo.scc.forwardBackward


Changed algo descriptions and docs to use `concurrency: 4`
